packages/samples/common-types/openapi/v5/networksecurityperimeter.json

```diff
@@ -0,0 +1,518 @@
+            "description": "Unknown issue type"
+          },
+          {
+            "name": "ConfigurationPropagationFailure",
+            "value": "ConfigurationPropagationFailure",
+            "description": "An error occurred while applying the network security perimeter (NSP) configuration."
+          },
+          {
+            "name": "MissingPerimeterConfiguration",
+            "value": "MissingPerimeterConfiguration",
+            "description": "A network connectivity issue is happening on the resource which could be addressed either by adding new resources to the network security perimeter (NSP) or by modifying access rules."
+          },
+          {
+            "name": "MissingIdentityConfiguration",
+            "value": "MissingIdentityConfiguration",
+            "description": "An managed identity hasn't been associated with the resource. The resource will still be able to validate inbound traffic from the network security perimeter (NSP) or matching inbound access rules, but it won't be able to perform outbound access as a member of the NSP."
+          }
+        ]
+      }
+    },
+    "NetworkSecurityPerimeter": {
+      "type": "object",
+      "description": "Information about a network security perimeter (NSP)",
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "arm-id",
+          "description": "Fully qualified Azure resource ID of the NSP resource",
+          "x-ms-arm-id-details": {
+            "allowedResources": [
+              {
+                "type": "Microsoft.Network/networkSecurityPerimeters"
+              }
+            ]
+          }
+        },
+        "perimeterGuid": {
+          "type": "string",
+          "format": "uuid",
+          "description": "Universal unique ID (UUID) of the network security perimeter"
+        },
+        "location": {
+          "type": "string",
+          "description": "Location of the network security perimeter",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      }
+    },
+    "NetworkSecurityPerimeterConfiguration": {
+      "type": "object",
+      "description": "Network security perimeter (NSP) configuration resource",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/NetworkSecurityPerimeterConfigurationProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "NetworkSecurityPerimeterConfigurationListResult": {
+      "type": "object",
+      "description": "Result of a list NSP (network security perimeter) configurations request.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "Array of network security perimeter results.",
+          "items": {
+            "$ref": "#/definitions/NetworkSecurityPerimeterConfiguration"
+          },
+          "x-ms-identifiers": []
```

Comment:
This wasn't there before
<img width="389" alt="image" src="https://github.com/user-attachments/assets/04de8215-4cb1-4781-b89c-62b4c944b059">

Reply:
I tried to address this one but I am unsure if it is a problem on my definition or when the generation of the common types is happening.

For what I saw, "x-ms-identifiers" is included when there is not an id for the array, in this case there is an id coming from resource: https://github.com/Azure/typespec-azure/blob/80f55767e0ff73699de4cba05b754529f8a5fc14/packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp#L170 (is not directly in there but it is pulled) even if I tried to add an id, I am not able to do so because it is already inherited from the other model, but when the common-type is getting generated it includes the x-ms-identifiers like it didn't have the id, am I missing something?

Reply:
I think this is related to this bug: https://github.com/Azure/typespec-azure/issues/1358  as the same logic is likely to be used

If so, we probably want to fix this before we merge this PR